### PR TITLE
test: add tests for test success

### LINKED ISSUE #6: Review test success (action failure) tests
Review test success (action failure) tests

### PR DESCRIPTION
**Pull request by ianlnf**
- Adds tests for succeeding tests resulting in action failure (as they do not test production changes)

Resolves #6

### CODE REVIEW

File: action.yml
Comment (ianlnf) [inline action.yml:64]: reverted as author cannot be specified when creating/updating comments, comments are filtered by body includes 'Test compare'

Comment (simoneb) [reply action.yml:64]: can you help me understand this? why would you be able to set one author but not the other? is there anything special about github-actions[bot]?

Comment (ianlnf) [reply action.yml:64]: This is the default author for comments created via custom GitHub Actions, I assume a GitHub App would be required to change that behaviour

Comment (simoneb) [reply action.yml:64]: ok that explains it. for the sake of future-proofness, nonetheless, I'd include in the comment some hidden data that we can use and that gives us stronger guarantees that we are finding the comment we're looking for, rather than one that by pure chance contains "test compare" (as unlikely as it may seem)

Comment (ianlnf) [reply action.yml:64]: Updated `body-includes` to full comment title

**Review by simoneb — COMMENTED** [review]
code wise LGTM, let's wait for a review from @RafaelGSS

File: action.yml
Comment (RafaelGSS) [inline action.yml:37]: What's ACT and when it will contain a value?

Comment (ianlnf) [reply action.yml:37]: The action is tested with act-js and Mock GitHub, this is an env var only available when running tests with Act. There are limitations with act-js/Mock GitHub, specifically it does not allow for mocking steps within a composite action (only workflows), hence the use of stubbed steps. There is a project idea for an improved GitHub tester that would allow for mocking action steps, and these `env.ACT` checks could then be removed.

**Review by simoneb — APPROVED** [review]

**Pull request merged**
